application/models/Campaign_model.php

```diff
@@ -1381,9 +1381,7 @@ public function archive_file($filetype, $office_id, $url)
 
         $context = stream_context_create($opts);
 
-        $copy = @fopen($url, 'rb', false, $context);
-        $paste = @fopen($filepath, 'wb');
-
+        $copy = @fopen(filter_remote_url($url), 'rb', false, $context);
```

Comment:
This is probably safe, but I think the `archive_file` is dead code that @pburkholder identified. I suspect the URL is being filtered earlier.

Reply:
From the README it seems like it's only dead under certain conditions related to S3. We added a test for it recently based on it causing problems during crawls, so it's definitely being invoked.